File: backend/middleware/response.js
/**
 * 统一响应格式中间件
 */
module.exports = function(req, res, next) {
  /**
   * 成功响应
   * @param {*} data - 业务数据
   * @param {string} message - 提示信息
   * @param {number} code - 业务状态码
   */
  res.success = function(data = null, message = '操作成功', code = 200) {
    return res.status(200).json({
      code,
      success: true,
      message,
      data,
      timestamp: new Date().toISOString()
    });
  };

  /**
   * 错误响应
   * @param {string} message - 错误信息
   * @param {number} code - 业务错误码
   * @param {number} httpStatus - HTTP状态码
   */
  res.error = function(message = '操作失败', code = 500, httpStatus = 500) {
    return res.status(httpStatus).json({
      code,
      success: false,
      message,
      data: null,
      timestamp: new Date().toISOString()
    });
  };

  /**
   * 分页响应
   * @param {Array} items - 数据列表
   * @param {number} total - 总数
   * @param {number} page - 当前页
   * @param {number} pageSize - 每页大小
   */
  res.paginate = function(items, total, page = 1, pageSize = 10) {
    return res.status(200).json({
      code: 200,
      success: true,
      message: '获取成功',
      data: {
        items,
        pagination: {
          total,
          page,
          pageSize,
          totalPages: Math.ceil(total / pageSize)
        }
      },
      timestamp: new Date().toISOString()
    });
  };

  next();
};

File: response.py
"""
统一响应格式工具
"""
from flask import jsonify
from datetime import datetime

def success_response(data=None, message="操作成功", code=200):
    """
    成功响应
    
    Args:
        data: 业务数据
        message: 提示信息
        code: 业务状态码
    
    Returns:
        JSON响应和HTTP状态码
    """
    return jsonify({
        "code": code,
        "success": True,
        "message": message,
        "data": data,
        "timestamp": datetime.utcnow().isoformat() + 'Z'
    }), 200

def error_response(message="操作失败", code=500, http_status=500):
    """
    错误响应
    
    Args:
        message: 错误信息
        code: 业务错误码
        http_status: HTTP状态码
    
    Returns:
        JSON响应和HTTP状态码
    """
    return jsonify({
        "code": code,
        "success": False,
        "message": message,
        "data": None,
        "timestamp": datetime.utcnow().isoformat() + 'Z'
    }), http_status

def paginate_response(items, total, page=1, page_size=10, message="获取成功"):
    """
    分页响应
    
    Args:
        items: 数据列表
        total: 总数
        page: 当前页
        page_size: 每页大小
        message: 提示信息
    
    Returns:
        JSON响应和HTTP状态码
    """
    import math
    
    return jsonify({
        "code": 200,
        "success": True,
        "message": message,
        "data": {
            "items": items,
            "pagination": {
                "total": total,
                "page": page,
                "pageSize": page_size,
                "totalPages": math.ceil(total / page_size) if page_size > 0 else 0
            }
        },
        "timestamp": datetime.utcnow().isoformat() + 'Z'
    }), 200
